src/seedu/addressbook/commands/Command.java

```diff
@@ -47,6 +47,13 @@ public CommandResult execute(){
     //public abstract CommandResult execute();
 
     /**
+     *
+     * @return true if command type mutates data
+     */
+    public boolean isMutating(){
+        return false;
+    }
```

Comment:
Instead of using `@return` you can just write it like `Returns true if …`

Having a default implementation for the `isMutating` method helps to reduce the amount of code. However, some may argue that choosing `false` (or `true`) as the default is arbitrary and hurts the intuitiveness of the code. Making the method `abstract` forces all child classes to declare whether it is a mutating command or not, rather than depend on an arbitrary decision made in the parent class.